diff --git a/cpython/Makefile b/cpython/Makefile
index cbdd45fa..d82f544f 100644
--- a/cpython/Makefile
+++ b/cpython/Makefile
@@ -12,10 +12,10 @@ URL=https://www.python.org/ftp/python/$(PYVERSION)/Python-$(PYVERSION).tgz
 LIB=libpython$(PYMAJOR).$(PYMINOR).a
 SYSCONFIG_NAME=_sysconfigdata__emscripten_
 
-ZLIBVERSION = 1.2.11
+ZLIBVERSION = 1.2.12
 ZLIBTARBALL=$(ROOT)/downloads/zlib-$(ZLIBVERSION).tar.gz
 ZLIBBUILD=$(ROOT)/build/zlib-$(ZLIBVERSION)
-ZLIBURL=https://zlib.net/zlib-1.2.11.tar.gz
+ZLIBURL=https://zlib.net/zlib-1.2.12.tar.gz
 
 SQLITETARBALL=$(ROOT)/downloads/sqlite-autoconf-3270200.tar.gz
 SQLITEBUILD=$(ROOT)/build/sqlite-autoconf-3270200
@@ -87,7 +87,7 @@ $(BUILD)/.patched: $(TARBALL)
 
 $(ZLIBBUILD)/.configured: $(ZLIBTARBALL)
 	[ -d $(ROOT)/build ] || (mkdir $(ROOT)/build)
-	tar -C $(ROOT)/build/ -xf $(ROOT)/downloads/zlib-1.2.11.tar.gz
+	tar -C $(ROOT)/build/ -xf $(ROOT)/downloads/zlib-1.2.12.tar.gz
 	cd $(ZLIBBUILD); emconfigure ./configure
 	touch $@
 
diff --git a/packages/zlib/meta.yaml b/packages/zlib/meta.yaml
index 0089f3f2..e1060f39 100644
--- a/packages/zlib/meta.yaml
+++ b/packages/zlib/meta.yaml
@@ -1,10 +1,10 @@
 package:
   name: zlib
-  version: 1.2.11
+  version: 1.2.12
 
 source:
-  sha256: c3e5e9fdd5004dcb542feda5ee4f0ff0744628baf8ed2dd5d66f8ca1197cb1a1
-  url: https://zlib.net/zlib-1.2.11.tar.gz
+  sha256: 91844808532e5ce316b3c010929493c0244f3d37593afd6de04f71821d5136d9
+  url: https://zlib.net/zlib-1.2.12.tar.gz
 
 build:
   library: true
